syntax = "proto3";

package bar_archive;

service BarArchiveService {
  rpc GetBeverageByID (GetBeverageByIDRequest) returns (GetBeverageByIDResponse);
  rpc AddBeverage (AddBeverageRequest) returns (AddBeverageResponse);
  rpc EditBeverage (EditBeverageRequest) returns (EditBeverageResponse);
  rpc RemoveBeverage (RemoveBeverageRequest) returns (RemoveBeverageResponse);

  rpc GetBeverages (GetBeveragesRequest) returns (GetBeveragesResponse);

}

message Beverage {
  int32 beverage_id = 1;
  string name = 2;
  string description = 3;
  string created_at = 4;
  string updated_at = 5;
}

message Ingredient {
  int32 ingredient_id = 1;
  int32 beverage_id = 2;
  string ingredient_name = 3;
  string created_at = 4;
  string updated_at = 5;
}

message Method {
  int32 method_id = 1;
  int32 beverage_id = 2;
  string description = 3;
  string created_at = 4;
  string updated_at = 5;
}

message GetBeveragesRequest {
}

message GetBeveragesResponse {
  repeated Beverage beverages = 1;
}

message GetBeverageByIDRequest {
  int32 beverage_id = 1;
}

message GetBeverageByIDResponse {
  repeated Beverage beverages = 1;
  Ingredient ingredient = 2;
  Method method = 3;
}

message AddBeverageRequest {
  Beverage beverages = 1;
  Ingredient ingredient = 2;
  Method method = 3;
}

message AddBeverageResponse {
  string message = 1;
}

message EditBeverageRequest {
  repeated Beverage beverages = 1;
  Ingredient ingredient = 2;
  Method method = 3;
}

message EditBeverageResponse {
  string message = 1;
}

message RemoveBeverageRequest {
  int32 beverage_id = 1;
}

message RemoveBeverageResponse {
  string message = 1;
}
